apps/files_sharing/src/components/NewFileRequestDialog/NewFileRequestDialogFinish.vue

```diff
@@ -0,0 +1,224 @@
+		</NcNoteCard>
+
+		<!-- Copy share link -->
+		<NcInputField ref="clipboard"
+			:value="shareLink"
+			:label="t('files_sharing', 'Share link')"
+			:readonly="true"
+			:show-trailing-button="true"
+			:trailing-button-label="t('files_sharing', 'Copy to clipboard')"
+			@click="copyShareLink"
+			@click-trailing-button="copyShareLink">
+			<template #trailing-button-icon>
+				<IconCheck v-if="isCopied" :size="20" @click="isCopied = false" />
+				<IconClipboard v-else :size="20" @click="copyShareLink" />
+			</template>
+		</NcInputField>
+
+		<template v-if="isShareByMailEnabled">
+			<!-- Email share-->
+			<NcTextField :value.sync="email"
+				:label="t('files_sharing', 'Send link via email')"
+				:placeholder="t('files_sharing', 'Enter an email address or paste a list')"
+				type="email"
+				@keypress.enter.stop="addNewEmail"
+				@paste.stop.prevent="onPasteEmails" />
+
+			<!-- Email list -->
+			<div v-if="emails.length > 0" class="file-request-dialog__emails">
+				<NcChip v-for="mail in emails"
+					:key="mail"
+					:aria-label-close="t('files_sharing', 'Remove email')"
+					:text="mail"
+					@close="$emit('remove-email', mail)">
+					<template #icon>
+						<NcAvatar :disable-menu="true"
+							:disable-tooltip="true"
+							:is-guest="true"
+							:size="24"
+							:user="mail" />
+					</template>
+				</NcChip>
+			</div>
+		</template>
+	</div>
+</template>
+
+<script lang="ts">
+import type { PropType } from 'vue'
+import Share from '../../models/Share'
+
+import { defineComponent } from 'vue'
+import { generateUrl } from '@nextcloud/router'
+import { showError, showSuccess } from '@nextcloud/dialogs'
+import { translate, translatePlural } from '@nextcloud/l10n'
+
+import NcAvatar from '@nextcloud/vue/dist/Components/NcAvatar.js'
+import NcInputField from '@nextcloud/vue/dist/Components/NcInputField.js'
+import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+import NcChip from '@nextcloud/vue/dist/Components/NcChip.js'
+
+import IconCheck from 'vue-material-design-icons/Check.vue'
+import IconClipboard from 'vue-material-design-icons/Clipboard.vue'
+
+export default defineComponent({
+	name: 'NewFileRequestDialogFinish',
+
+	components: {
+		IconCheck,
+		IconClipboard,
+		NcAvatar,
+		NcInputField,
+		NcNoteCard,
+		NcTextField,
+		NcChip,
+	},
+
+	props: {
+		share: {
+			type: Object as PropType<Share>,
+			required: true,
+		},
+		emails: {
+			type: Array as PropType<string[]>,
+			required: true,
+		},
+		isShareByMailEnabled: {
+			type: Boolean,
+			required: true,
+		},
+	},
+
+	emits: ['add-email', 'remove-email'],
+
+	setup() {
+		return {
+			n: translatePlural,
+			t: translate,
+		}
+	},
+
+	data() {
+		return {
+			isCopied: false,
+			email: '',
+		}
+	},
+
+	computed: {
+		shareLink() {
+			return window.location.protocol + '//' + window.location.host + generateUrl('/s/') + this.share.token
```

Comment:
```suggestion
			return generateUrl('/s/{token}', { token: this.share.token }, { baseURL: getBaseUrl() })
```

(`getBaseUrl` from `@nextcloud/router`)